ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -628,38 +652,37 @@ static bool ShouldHandleFileOpEvent(
     kauth_action_t action,
 
     // Out params:
-    VirtualizationRoot** root,
+    int16_t* rootIndex,
     int* pid)
 {
+    *rootIndex = RootIndex_None;
+
     vtype vnodeType = vnode_vtype(vnode);
     if (ShouldIgnoreVnodeType(vnodeType, vnode))
     {
         return false;
     }
 
-    *root = VirtualizationRoots_FindForVnode(vnode);
-    if (nullptr == *root)
-    {
-        return false;
-    }
-    else if (nullptr == (*root)->providerUserClient)
+    *rootIndex = VirtualizationRoot_FindForVnode(vnode);
+
+    if (RootIndex_None >= *rootIndex)
```

Comment:
This check seems pretty magical, and it really assumes knowledge of the fact that these are indexes (which I think we should keep private). Make this a helper function instead?

Reply:
Sounds sensible, I've created a `VirtualizationRoot_IsValidRootHandle()` helper function instead.